File: 编译原理+操作系统+图形学/TinyScript-js/Token.js
const TokenType = require('./lexer/TokenType')
class Token {
  constructor(type, value) {
    this._type = type
    this._value = value
  }

  getType() {
    return this._type
  }

  isVariable() {
    return this._type === TokenType.VARIABLE
  }

  isSCalar() {
    return this._type === TokenType.INTEGER ||
        this._type === TokenType.FLOAT ||
        this._type === TokenType.STRING ||
        this._type === TokenType.BOOLEAN
  }

  toStirng() {
    return `type ${this._type.type}, value ${this.value}`
  }
}

module.exports = Token

File: 编译原理/TinyScript/src/main/java/lexer/Token.java
package lexer;

public class Token {
    TokenType _type;
    String _value;

    public Token(TokenType type, String value) {
        this._type = type;
        this._value = value;
    }

    public TokenType getType() {
        return  _type;
    }

    @Override
    public String toString() {
        return String.format("type %s, value %s", _type, _value);
    }

    /**
     * 是否是变量
     *
     * @return boolean
     */
    public boolean isVariable() {
        return _type == TokenType.VARIABLE;
    }

    /**
     * 是否是值类型
     *
     * @return boolean
     */
    public boolean isScalar() {
        return _type == TokenType.INTEGER || _type == TokenType.FLOAT ||
                _type == TokenType.STRING || _type == TokenType.BOOLEAN;
    }
}
